src/utils/encoding.ts

```diff
@@ -0,0 +1,71 @@
+import { assertUint64 } from "./number";
+import { stringToUint8ArrayUtf8 } from "./string";
+
+/**
+ * Converts the given number into a uint8 array
+ *
+ * @param num - Number to encode.
+ * @returns - Number encoded as a byte array.
+ */
+export function encodeNumber(num: number): Uint8Array {
+  const encoded = new Uint8Array(8);
+  for (let index = 0; index < encoded.length; index++) {
+    const byte = num & 0xff;
+    encoded[index] = byte;
+    num = num >> 8;
+  }
+  return encoded;
+}
+
+/**
+ * Encodes the given bigint into a uint8 array.
+ *
+ * @param int - Bigint to encode.
+ * @returns - Bigint encoded as a byte array.
+ * @throws - Will throw if the int does not fit in 64 bits.
+ */
+export function encodeBigintAsUint64(int: bigint): Uint8Array {
+  // Assert the input is 64 bits.
+  assertUint64(int);
+
+  const encoded = new Uint8Array(8);
+  for (let index = 0; index < encoded.length; index++) {
+    const byte = int & BigInt(0xff);
+    encoded[index] = Number(byte);
+    int = int >> BigInt(8);
+  }
+  return encoded;
+}
+
+/**
+ * Encodes the uint8array, prefixed by its length.
+ *
+ * @param bytes - The input array.
+ * @returns - The encoded byte array.
+ */
+export function encodePrefixedBytes(bytes: Uint8Array): Uint8Array {
+  const len = bytes.length;
+  const buf = new ArrayBuffer(8 + len);
+  const view = new DataView(buf);
+
+  // Sia uses setUint64 which is unavailable in JS.
```

Comment:
don't you have `setBigUint64`?

Reply:
Not available in Safari 😬 When I last needed this for adding skykeys (which we didn't go through this) I used some bigint library, but setUint32 works and `len` will never be that big.

Reply:
works for me